.github/workflows/check-conventional-commits.yml

```diff
@@ -0,0 +1,119 @@
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    outputs:
+      raw_output: ${{ steps.cc-lint.outputs.raw_output }}
+      raw_errors: ${{ steps.cc-lint.outputs.raw_errors }}
+      json_output: ${{ steps.cc-lint.outputs.json_output }}
+    steps:
+      - name: Action Meta
+        id: meta
+        run: |
+          # build action meta
+          echo "fetch_depth=$(( ${{ github.event.pull_request.commits }} + 100 ))" >> "$GITHUB_OUTPUT"
+
+      - name: Checkout PR branch & commits
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: ${{ steps.meta.outputs.fetch_depth }}
+
+      - name: Fetch Base commit
+        run: |
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+
+      - name: Install convco
+        uses: taiki-e/install-action@77b010ce7843a10f653ccb6b0967bda4c3d81091 # v2.47.1
+        with:
+          tool: convco@0.6.1
+
+      - name: Check commits for formatting
+        id: cc-lint
+        run: |
+          set +e
+          OUTPUT=$(convco check ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }})
+          set -e
+
+          # output results
+          if [[ $OUTPUT =~ FAIL ]]; then
+            echo '---- errors found ----'
+            RAW_ERRORS=$(echo "$OUTPUT" | grep -x '^FAIL.*$')
+            echo "$RAW_ERRORS"
+            JSON_OUTPUT=$(echo "$RAW_ERRORS" | jq -R -c 'split("\n") | map(split("  ") | { "key": .[1], "value": .[2] }) | { "commits": from_entries, "errors": length }')
+            echo '-------- json --------'
+            echo "$JSON_OUTPUT"
+            echo '----------------------'
+            {
+              echo "json_output=$JSON_OUTPUT"
+              echo "raw_output<<EOF"
+              echo "$OUTPUT"
+              echo "EOF"
+              echo "raw_errors=$RAW_ERRORS"
+            } >> "$GITHUB_OUTPUT"
+            exit 1
+          else
+            echo '---- no errors found ----'
+            {
+              echo "json_output=$(jq -n -c '{ "commits": {}, "count": 0 }')" >> "$GITHUB_OUTPUT"
+              echo 'raw_errors=""'
+              echo "raw_output<<EOF" 
+              echo "$OUTPUT"
+              echo "EOF"
+            } >> "$GITHUB_OUTPUT"
+            exit 0
+          fi
+
+      - name: Job Summary
+        if: always()
+        env:
+          JOB_OUTPUT: ${{ steps.cc-lint.outputs.raw_output }}
+          JOB_STATUS: ${{ steps.cc-lint.outcome == 'success' && '✅' || steps.cc-lint.outcome == 'failure' && '❌' || steps.cc-lint.outcome == 'cancelled' && '🚫' }}
+        run: |
+          {
+            echo "## $JOB_STATUS Conventional Commits"
+            echo '```'
+            echo "$JOB_OUTPUT"
+            echo '```'
+          } >> $GITHUB_STEP_SUMMARY
+
+  comment:
+    if: always() && (needs.lint.result == 'success' || needs.lint.result == 'failure')
```

Comment:
Is the `always()` here necessary?

Reply:
Yeah, it feels weird but `always()` is needed for eager evaluation of the if condition when `needs` is defined. [See this run](https://github.com/wasmCloud/wasmCloud/actions/runs/12587027975/job/35082058769?pr=3911).